public/app/features/variables/datasource/actions.ts

```diff
@@ -27,16 +30,21 @@ export const updateDataSourceVariableOptions = (
     regex = stringToJsRegex(regex);
   }
 
-  await dispatch(createDataSourceOptions(toVariablePayload(identifier, { sources, regex })));
+  dispatch(createDataSourceOptions(toVariablePayload(identifier, { sources, regex })));
   await dispatch(validateVariableSelectionState(identifier));
 };
 
 export const initDataSourceVariableEditor = (
   dependencies: DataSourceVariableActionDependencies = { getDatasourceSrv: getDatasourceSrv }
-): ThunkResult<void> => async dispatch => {
-  const dataSources: DataSourceSelectItem[] = await dependencies.getDatasourceSrv().getMetricSources();
-  const filtered = dataSources.filter(ds => !ds.meta.mixed && ds.value !== null);
-  const dataSourceTypes = _(filtered)
+): ThunkResult<void> => dispatch => {
+  const dataSources: DataSourceSelectItem[] = dependencies
+    .getDatasourceSrv()
+    .getList({ metrics: true, variables: true })
+    .filter(setting => {
+      return !setting.meta.mixed && !setting.isDefault;
```

Comment:
no need for this, mixed is opt-in, and isDefault is just a property on the option, so should filter out the default data source, as then you only have one data source and that is the default then this dropdown will be empty. (the old getMetricSources returned two entries, one named default with value null, getList does not do this, just returns the data sources, and if it happens to be the default isDefault is true)

Reply:
Great explanation, I'll remove the filter and update the tests.